client/go/src/github.com/vespa-engine/vespa/cmd/command_tester.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// A helper for testing commands
+// Author: bratseth
+
+package cmd
+
+import (
+    "bytes"
+    "github.com/vespa-engine/vespa/utils"
+    "github.com/stretchr/testify/assert"
+	"io/ioutil"
+    "net/http"
+    "testing"
+    "time"
+)
+
+// The HTTP status code that will be returned from the next invocation. Default: 200
+var nextStatus int
+
+// The response body code that will be returned from the next invocation. Default: ""
+var nextBody string
+
+// A recording of the last HTTP request made through this
+var lastRequest *http.Request
+
+func reset() {
+    // Persistent flags in Cobra persists over tests
+	rootCmd.SetArgs([]string{"status", "-t", ""})
+	rootCmd.Execute()
+
+    lastRequest = nil
+    nextStatus = 200
+    nextBody = ""
+}
+
+func executeCommand(t *testing.T, args []string, moreArgs []string) (standardout string) {
+    utils.ActiveHttpClient = mockHttpClient{}
+	b := bytes.NewBufferString("")
+    utils.Out = b
+	rootCmd.SetArgs(concat(args, moreArgs))
+	rootCmd.Execute()
+	out, err := ioutil.ReadAll(b)
+	assert.Empty(t, err, "No error")
+	return string(out)
+}
+
+type mockHttpClient struct {
```

Comment:
Instead of having package level fields, `lastRequest` and friends can be moved to this struct.

Reply:
That requires passing the client explicitly. A bit more verbose, but also less magical, so: done.